src/oauth/consumer.cr

```diff
@@ -63,7 +63,8 @@ class OAuth::Consumer
                  @request_token_uri : String = "/oauth/request_token",
                  @authorize_uri : String = "/oauth/authorize",
                  @access_token_uri : String = "/oauth/access_token",
-                 @tls : Bool | OpenSSL::SSL::Context::Client = @scheme == "https")
+                 @tls : Bool | OpenSSL::SSL::Context::Client = false)
+    @tls = @scheme == "https" unless @tls.is_a?(OpenSSL::SSL::Context::Client)
```

Comment:
@RX14: Check the commit log, i specifically explain that the logic here is not sound because of the Union handling problem when calling HTTP::Client.new with using the three member Union of _Nil | Bool | OpenSSL::SSL::Context::Client_ since HTTP::Client's constructor expects a Union with only _Bool | OpenSSL::SSL::Context::Client_ members. I'm pretty sure this is a compiler level error - my original implementation was clean and retained logic consistency for assigning @tls. I will note that today you simply _cant_ set @tls to false, so while the condition we both noticed is problematic for good programmers (this is how bugs and security flaws are born), it doesn't cause a functional regression, and should be fixed by allowing the original design of this PR to work in the initializer's arg processing stack.
@Sija: This is a positive check, for the OpenSSL::SSL::Context::Client member in the Union.
@all: So the hack here would be something like 
```
...
    @tls_ctx : Nil | OpenSSL::SSL::Context::Client = nil)
  @tls = @tls_ctx.nil? ? @scheme == "https" : @tlx_ctx
...
```
Sorry for the eye sore, but given the logic constraint, union type check, and initializer arg misbehavior, this might have to be the "safe until required functionality is in place lower in the stack" approach